praw/models/util.py

```diff
@@ -77,12 +77,26 @@ def permissions_string(permissions, known_permissions):
     return ','.join(to_set)
 
 
-def stream_generator(function):
-    """Forever yield new items from ListingGenerators."""
+def stream_generator(function, pause_after=None):
+    """Forever yield new items from ListingGenerators.
+
+    :param function: A callable that returns a ListingGenerator, e.g.
+       ``subreddit.comments`` or ``subreddit.new``.
+    :param pause_after: A integer representing the maximum times to fetch
+       data if no new items are found. Once this limit is reached, it will
+       yield ``None``, signalling to the caller no new data is available.
+       If set to ``0``, it will yield ``None`` after each data fetch that
+       returns no new items. If not set, the fetching loop will just keep
+       running (with sleeps) until a valid item appears. Once ``None`` is
+       yielded, the calling code can still consume the generator, which will
+       yield new data once available using the same ``pause_after`` value
+       (default: None).
+    """
     before_fullname = None
     exponential_counter = ExponentialCounter(max_counter=64)
     seen_fullnames = BoundedSet(301)
     without_before_counter = 0
+    failed_fetches = 0
```

Comment:
Changed to `responses_without_new` and also stopped checking that pause_after is None on every iteration.